crates/storage_impl/src/payments/payment_intent.rs

```diff
@@ -600,6 +601,23 @@ impl<T: DatabaseStore> PaymentIntentInterface for crate::RouterStore<T> {
 
                 query = query.offset(params.offset.into());
 
+                if let Some(amount_filter) = &params.amount {
+                    match &amount_filter.filter_option {
+                        AmountFilterOption::LessThan => {
+                            query = query.filter(pi_dsl::amount.lt(amount_filter.amount))
+                        }
+                        AmountFilterOption::EqualTo => {
+                            query = query.filter(pi_dsl::amount.eq(amount_filter.amount))
+                        }
+                        AmountFilterOption::GreaterThan => {
+                            query = query.filter(pi_dsl::amount.gt(amount_filter.amount))
+                        }
+                        AmountFilterOption::Range { end } => {
+                            query = query.filter(pi_dsl::amount.between(amount_filter.amount, *end))
```

Comment:
How will this query work if `end` is less than `amount`?

Reply:
It will give empty list